src/ape/plugins/query.py

```diff
@@ -0,0 +1,27 @@
+from typing import TYPE_CHECKING, Iterator, Tuple, Type
+
+from .pluggy_patch import PluginType, hookspec
+
+if TYPE_CHECKING:
+    from ape.api import QueryAPI
+
+
+class QueryPlugin(PluginType):
+    """
+    A plugin for querying chains.
+    """
+
+    @hookspec
+    def query_engines(self) -> Iterator[Tuple[str, Type["QueryAPI"]]]:
```

Comment:
Since `plugin_name` is already given by the PluginManager, this would be redundant (e.g. `ape-postgres` already tells you it's `postgres`)
```suggestion
    def query_engines(self) -> Iterator[Type["QueryAPI"]]:
```